### PR TITLE
Add release notes and highlights for 2.5.0

### PR DESCRIPTION
**Pull request by pebrc**
Preview https://cloud-on-k8s_6069.docs-preview.app.elstc.co/diff

### CODE REVIEW

**Comment by pebrc** [comment]
@elasticmachine, run elasticsearch-ci/docs

File: docs/release-notes/2.5.0.asciidoc [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+
+[[feature-2.5.0]]
+[float]
+=== New features
+
+* Autoscaling Elasticsearch: Introduce a dedicated custom resource {pull}5978[#5978] (issue: {issue}5997[#5997])
+* ECK resources Helm chart - Elastic Agent & Elastic Fleet Server Agent {pull}5889[#5889] (issue: {issue}5505[#5505])
+* Enable beats stack monitoring configuration {pull}5878[#5878] (issue: {issue}5563[#5563])
```
Comment (barkbay) [inline docs/release-notes/2.5.0.asciidoc:15]: nit: I think the first letter should be capitalized
```suggestion
* Enable Beats stack monitoring configuration {pull}5878[#5878] (issue: {issue}5563[#5563])
```

File: docs/release-notes/highlights-2.5.0.asciidoc [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+==== Autoscaling Elasticsearch: Introducing a dedicated custom resource
+
+The ECK operator has had experimental support for autoscaling Elasticsearch since version 1.5.0. With this release we are removing the experimental label and giving the autoscaling specification its own custom resource definition with improved status reporting compared to the experimental implementation. Read our updated <<{p}-autoscaling, docs>> for more details.
+
+
+[float]
+[id="{p}-250-agent-fleet-helm-chart"]
+==== Helm charts for Elastic Agent and Fleet Server
+
+We are adding new two new Helm charts for workloads managed by the ECK operator. Charts for Elastic Agent and Fleet Server are complementing the existing charts for Elasticsearch and Kibana. They can be used individually or as part of our existing Elastic Stack Helm chart. Get started by reading the  <<{p}-stack-helm-chart, docs>>.
```
Comment (barkbay) [inline docs/release-notes/highlights-2.5.0.asciidoc:21]: nit: duplicated "new"?
```suggestion
We are adding two new Helm charts for workloads managed by the ECK operator. Charts for Elastic Agent and Fleet Server are complementing the existing charts for Elasticsearch and Kibana. They can be used individually or as part of our existing Elastic Stack Helm chart. Get started by reading the  <<{p}-stack-helm-chart, docs>>.
```

**Review by barkbay — APPROVED** [review]
LGTM

File: docs/release-notes/2.5.0.asciidoc [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+* Increment desired nodes version on each call {pull}6037[#6037] (issue: {issue}5979[#5979])
+* Ignore unmanaged namespaces in webhook validation for all resources. {pull}6013[#6013] (issue: {issue}5814[#5814])
+* Fix helm chart rendering issues associated with indentation {pull}6004[#6004]
+* Stack monitoring: trust custom cert. in output configuration {pull}5945[#5945] (issue: {issue}5917[#5917])
+
+[[docs-2.5.0]]
+[float]
+=== Documentation improvements
+
+* Adding License files for Helm Charts and Updating Chart README {pull}6008[#6008] (issue: {issue}6005[#6005])
```
Comment (alaudazzi) [inline docs/release-notes/2.5.0.asciidoc:43]:
```suggestion
* Add License files for Helm Charts and Updating Chart README {pull}6008[#6008] (issue: {issue}6005[#6005])
```
just to stay consistent with the other entries

**Review by alaudazzi — APPROVED** [review]
A very minor suggestion, otherwise LGTM.

**Comment by pebrc** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|2.5|[](https://github.com/elastic/cloud-on-k8s/pull/6085)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**